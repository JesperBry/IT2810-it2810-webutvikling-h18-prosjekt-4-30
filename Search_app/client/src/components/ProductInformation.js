import React, { Component } from "react";
import "../styles/ProductInformation.css";

export default class ProductInformation extends Component {
  render() {
    const product = this.props.productItem;
    return (
      <div className={"infoContainer"}>
        <div className={"row1"}>
          <h4 className={"infoHeader"}>Info</h4>
          <p>Volum: {product.Volum}</p>
          <p>Alkohol: {product.Alkohol}%</p>
          <p>Land: {product.Land}</p>
        </div>
        <div className={"row2"}>
          <h4 className={"infoHeader"}>Pris</h4>
          <p>
            Pris: {product.Pris}
            ,-
          </p>
          <p>
            Literpris: {product.Literpris}
            ,-
          </p>
          <p>
            Pris per alkoholenhet:{" "}
            {(product.Literpris / (product.Alkohol / 100) / (200 / 3)).toFixed(
              2
            )}
            {""}
            ,-
          </p>
        </div>
        <div className={"row3"}>
          <h4 className={"infoHeader"}>Smak</h4>
        </div>
      </div>
    );
  }
}
