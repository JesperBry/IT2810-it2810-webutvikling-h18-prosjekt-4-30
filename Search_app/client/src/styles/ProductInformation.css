@import url("https://fonts.googleapis.com/css?family=Roboto:300,400,500,700");

.paragraph {
  font-family: "Roboto", sans-serif;
}

.infoContainer {
  font-family: "Roboto", sans-serif;
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding-bottom: 5rem;
  margin-top: 3rem;
}

.row1 {
  padding-left: 5rem;
}

.row2 {
  padding-left: 10rem;
}

.infoHeader {
  padding-bottom: 1rem;
}
