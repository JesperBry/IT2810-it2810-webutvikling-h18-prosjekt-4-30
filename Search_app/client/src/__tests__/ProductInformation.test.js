import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Enzyme, { shallow } from "enzyme";

import ProductInformation from "../components/ProductInformation";

it("Matches snapshot", () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(
    <ProductInformation
      productItem={[
        {
          Volum: 0.7,
          Pris: 325.9,
          Literpris: 465.57,
          Farge: "Vannklar.",
          Lukt: "Ren og nøytral aroma.",
          Smak: "Lett. ren og nøytral drikk.",
          Land: "Sverige"
        }
      ]}
    />
  );
  expect(result).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(
    <ProductInformation
      productItem={[
        {
          Volum: 0.7,
          Pris: 325.9,
          Literpris: 465.57,
          Farge: "Vannklar.",
          Lukt: "Ren og nøytral aroma.",
          Smak: "Lett. ren og nøytral drikk.",
          Land: "Sverige"
        }
      ]}
    />
  );
});
