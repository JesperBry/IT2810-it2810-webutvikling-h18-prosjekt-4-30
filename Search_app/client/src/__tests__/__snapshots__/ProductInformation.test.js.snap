// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="paragraph"
>
  <p>
    <strong>
      Smak:
    </strong>
     
  </p>
  <p>
    <strong>
      Lukt:
    </strong>
     
  </p>
  <p>
    <strong>
      Farge:
    </strong>
     
  </p>
  <div
    className="infoContainer"
  >
    <div
      className="row1"
    >
      <h4
        className="infoHeader"
      >
        Info
      </h4>
      <p>
        Volum: 
         liter
      </p>
      <p>
        Alkohol: 
         %
      </p>
      <p>
        Land: 
      </p>
    </div>
    <div
      className="row2"
    >
      <h4
        className="infoHeader"
      >
        Pris
      </h4>
      <p>
        Pris: 
         ,-
      </p>
      <p>
        Literpris: 
         ,-
      </p>
      <p>
        Pris per alkoholenhet:
         
        NaN
        
         ,-
      </p>
    </div>
  </div>
</div>
`;
